tiledb/cc/CMakeLists.txt

```diff
@@ -33,6 +33,12 @@ target_compile_features(
     cxx_std_20
 )
 
+target_compile_definitions(
```

Comment:
Do we want to have this conditional based on some CMAKE variable like `-D TILEDB_REMOVE_DEPRECATIONS`?

Reply:
I mean if this is set here we don't need https://github.com/TileDB-Inc/TileDB-Py/blob/dev/pyproject.toml#L82 anymore